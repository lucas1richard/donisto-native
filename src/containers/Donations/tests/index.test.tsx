import * as React from 'react';
import { shallow } from 'enzyme';

import { Donations, IDonationsProps } from '../';

describe('<Donations />', () => {
  const renderComponent = (props: IDonationsProps) => shallow(<Donations {...props} />);
  it('Expect to have unit tests specified', () => {
    expect(renderComponent({
      donations: [],
      viewDetail: jest.fn(() => undefined)
    })).toMatchSnapshot();
  });
});
