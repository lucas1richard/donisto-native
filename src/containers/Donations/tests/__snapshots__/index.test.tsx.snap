// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Donations /> Expect to have unit tests specified 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Donations
    donations={Array []}
    viewDetail={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <DonationsHeader />
        <ScrollViewMock>
          <Component
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "padding": 7.5,
              }
            }
          />
        </ScrollViewMock>
      </Component>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <DonationsHeader />,
          <ScrollViewMock>
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "padding": 7.5,
                }
              }
            />
          </ScrollViewMock>,
        ],
        "style": Object {
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "padding": 7.5,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [],
              "style": Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "padding": 7.5,
              },
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <DonationsHeader />
          <ScrollViewMock>
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "padding": 7.5,
                }
              }
            />
          </ScrollViewMock>
        </Component>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <DonationsHeader />,
            <ScrollViewMock>
              <Component
                style={
                  Object {
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "padding": 7.5,
                  }
                }
              />
            </ScrollViewMock>,
          ],
          "style": Object {
            "height": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Component
                style={
                  Object {
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "padding": 7.5,
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "style": Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "padding": 7.5,
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
            "type": [MockFunction],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
